pkg/api/resolver_mutation_metadata.go

```diff
@@ -89,3 +93,41 @@ func (r *mutationResolver) JobStatus(ctx context.Context) (*models.MetadataUpdat
 func (r *mutationResolver) StopJob(ctx context.Context) (bool, error) {
 	return manager.GetInstance().Status.Stop(), nil
 }
+
+func (r *mutationResolver) BackupDatabase(ctx context.Context, input models.BackupDatabaseInput) (*string, error) {
+	// if download is true, then backup to temporary file and return a link
+	download := input.Download != nil && *input.Download
+	mgr := manager.GetInstance()
+	var backupPath string
+	if download {
+		utils.EnsureDir(mgr.Paths.Generated.Downloads)
+		f, err := ioutil.TempFile(mgr.Paths.Generated.Downloads, "backup*.sqlite")
+		if err != nil {
+			return nil, err
+		}
+
+		backupPath = f.Name()
+		f.Close()
+	} else {
+		backupPath = database.DatabaseBackupPath()
+	}
+
+	err := database.Backup(database.DB, backupPath)
+	if err != nil {
+		return nil, err
+	}
+
+	if download {
+		downloadHash := mgr.DownloadStore.RegisterFile(backupPath, "", false)
+		logger.Debugf("Generated backup file %s with hash %s", backupPath, downloadHash)
+
+		baseURL, _ := ctx.Value(BaseURLCtxKey).(string)
+
+		ret := baseURL + "/downloads/" + downloadHash + "/" + database.DatabaseBackupPath()
```

Comment:
While the plain  backup works the download one doesnt seem to.
Something seems wrong with the translation of the filename to the download path
```bash
INFO[0775] Backing up database into: /opt/nvme/stash/meta/downloads/backup199153937.sqlite 
DEBU[0780] Generated backup file /opt/nvme/stash/meta/downloads/backup199153937.sqlite with hash 9cb3394b 
```
the backup is correctly stored in my downloads folder but 
`http://localhost:9998/downloads/9cb3394b//home/username/.stash/stash-go.sqlite.17.20210120_144604`
is the link that is generated and 404s. 

i think in
`ret := baseURL + "/downloads/" + downloadHash + "/" + database.DatabaseBackupPath()`
the  `database.DatabaseBackupPath()` feels a little off, shouldnt it be `backupPath` instead?